documentation/src/main/wiki/Adapter.md

```diff
@@ -28,39 +30,150 @@ is not going to provide any generic Action Adapters.
 
 For custom Knots we can introduce custom Adapter types. As far as Knots must follow [[Knot contract|Knot#how-does-it-work]],
 Adapters are coupled with Knot directly so they can define their custom request, response or 
-configuration. The communication between Knot and Adapter can be custom too. 
+configuration. The communication between Knot and Adapter can be custom too.
+
+Communication contract between [[Service Knot|ServiceKnot]] and [[Http Service Adapter|HttpServiceAdapter]] is defined by:
+- `AdapterRequest` input,
+- `AdapterResponse` output.
+
+#### Adapter Request
+The table below describes the communication model between Knot.x Service Knot and Service Adapters.
+
+| Name                        | Type           | Mandatory | Description  |
+|-------:                     |:-------:       |:-------:   |-------|
+| `clientRequest.path`        | `String`       | &#10004;   | client request url |
+| `clientRequest.method`      | `HttpMethod`   | &#10004;   | client request method |
+| `clientRequest.headers`     | `MultiMap`     | &#10004;   | client request headers |
+| `clientRequest.params`      | `MultiMap`     | &#10004;   | client request parameters |
+| `params`                    | `JsonObject`   | &#10004;   | `JsonObject` with additional params that can be passed via configuration file |
+| `adapterParams`             | `JsonObject`   |            |  `JsonObject` with additional adapter parameters that can be set in the form of `data-knotx-adapter-params` in the snippet |
+
+#### Adapter Response
+The table below describes the result that Service Adapter returns to Service Knot.
```

Comment:
The table below shows all the fields in the AdapterResponse object returned by the Adapter to the Service Knot